pioneer/packages/joy-utils/src/transport/tokenomics.ts

```diff
@@ -341,4 +350,125 @@ export default class TokenomicsTransport extends BaseTransport {
+
+    return {
+      blockHeight,
+      numberOfMembers,
+      content: content.length,
+      numberOfChannels,
+      proposalCount,
+      historicalProposals: HISTORICAL_PROPOSALS.length,
+      numberOfForumCategories,
+      numberOfForumPosts,
+      councilMintCapacity: councilMint.capacity.toNumber(),
+      councilMintSpent: councilMint.total_minted.toNumber(),
+      contentCuratorMintCapacity: contentMint.capacity.toNumber(),
+      contentCuratorMintSpent: contentMint.total_minted.toNumber(),
+      storageProviderMintCapacity: storageMint.capacity.toNumber(),
+      storageProviderMintSpent: storageMint.total_minted.toNumber()
+    };
+  }
+
+  async proposalStatistics () {
+    const proposal = {
+      all: 0,
+      Active: 0,
+      Approved: 0,
+      Rejected: 0,
+      Expired: 0,
+      Slashed: 0,
+      Canceled: 0,
+      Vetoed: 0
+    };
+
+    const returnData = {
+      text: { ...proposal },
+      spending: { ...proposal },
+      workingGroups: { ...proposal },
+      networkChanges: { ...proposal },
+      all: { ...proposal }
+    };
+
+    const proposals = await this.entriesByIds<ProposalId, ProposalDetails>(this.api.query.proposalsCodex.proposalDetailsByProposalId);
+    const proposalData = await this.api.query.proposalsEngine.proposals.multi<Proposal>(proposals.map(([id, _]) => id));
+
+    proposalData.map((proposal, index) => {
+      const proposalType = proposals[index][1].type;
+      let definedProposalType: 'text' | 'spending' | 'networkChanges' | 'workingGroups';
+      const proposalStatus = proposal.status.isOfType('Active') ? 'Active' as const : proposal.status.asType('Finalized').proposalStatus.type;
+
+      if (proposalType === 'Text') {
+        definedProposalType = 'text';
+      } else if (proposalType === 'Spending') {
+        definedProposalType = 'spending';
+      } else if (proposalType === 'RuntimeUpgrade' || proposalType === 'SetElectionParameters' || proposalType === 'SetValidatorCount') {
+        definedProposalType = 'networkChanges';
+      } else {
+        definedProposalType = 'workingGroups';
+      }
+
+      returnData[definedProposalType].all++;
+      returnData[definedProposalType][proposalStatus]++;
+      returnData.all.all++;
+      returnData.all[proposalStatus]++;
+    });
+
+    return returnData;
+  }
+
+  historicalProposalStatistics () {
+    const proposal = {
+      all: 0,
+      Active: 0,
+      Approved: 0,
+      Rejected: 0,
+      Expired: 0,
+      Slashed: 0,
+      Canceled: 0,
+      Vetoed: 0
+    };
+
+    const historicalProposalData = {
+      text: { ...proposal },
+      spending: { ...proposal },
+      workingGroups: { ...proposal },
+      networkChanges: { ...proposal },
+      all: { ...proposal }
+    };
+
+    HISTORICAL_PROPOSALS.map(({ proposal }) => {
+      let proposalType: 'text' | 'spending' | 'networkChanges' | 'workingGroups';
+      const proposalStatus = proposal.status.Finalized
```

Comment:
You could use `createType('ProposalStatus', proposal.status)` and treat it the same way as current proposal statuses but as mentioned before, maybe this way is actually more furute-proof in case there will be some backward-incompatible changes in the runtime (though that would cause lots of other issues in Pioneer)

Reply:
I'll answer for both this one and the last comment on here as they are related. In my opinion due to `maybe this way is actually more furute-proof in case there will be some backward-incompatible changes in the runtime ` I would rather keep it as is. If you've changed your mind in the meantime and think we should try to implement it all in one function let me know!